main/src/library/InvalidKey.flix

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// Indicates that a key was invalid.
+/// Useful for instances of `Index` or `IndexMut`.
+///
+eff InvalidKey {
```

Comment:
The key is not invalid. Rather it does not exist: I would propose we call it either "NoSuchElement" or "InvalidIndex" or "IndexOutOfBounds". I think I prefer the last as the most precise.

Reply:
You're reading it as "key is not valid (forever)" but I think of it as "key is not valid (right now)". I think youre right that `InvalidKey` would be a better fit, compared to `InvalidIndex` if it was also called `Mapping` and `MappingMut`.

I don't think out of bounds is good, it sounds nonsense when you're working on `Map` fx

Another option is to have both `OutOfBounds` and `InvalidIndex`.

Reply:
For Java familiarity, I think OutOfBounds would be most understandable?

Reply:
`OutOfBounds` feels very weird to me for maps. What are the bounds?

Reply:
The set of keys. Its just a collection of discrete bounds.